src/mark.c

```diff
@@ -141,6 +143,17 @@ gboolean r_mark_active(RaucSlot *slot, GError **error)
 	r_slot_status_load(slot);
 	slot_state = slot->status;
 
+	/* It is more sensible to unlock the counter, before marking it primary. Reversing the order would create
```

Comment:
```suggestion
	/* It is more sensible to unlock the counter before marking it primary. Reversing the order would create
```

For users not aware of the previous implementation attempt, the 'It is more sensible' sounds a bit unexpected. We also don't mark the counter primary.
So maybe just start with `Unlock boot counting before marking the slot primary...`.